validator/client/beacon-api/prysm_beacon_chain_client.go

```diff
@@ -72,3 +75,39 @@ func (c prysmChainClient) ValidatorCount(ctx context.Context, stateID string, st
 
 	return resp, nil
 }
+
+func (c prysmChainClient) ValidatorPerformance(ctx context.Context, in *ethpb.ValidatorPerformanceRequest) (*ethpb.ValidatorPerformanceResponse, error) {
+	// Check node version for prysm beacon node as it is a custom endpoint for prysm beacon node.
+	nodeVersion, err := c.nodeClient.Version(ctx, nil)
```

Comment:
having this pattern and calling it here really sucks... we should really have this in global state based on what we are initially connected to.